Makefile.js

```diff
@@ -1037,172 +1055,90 @@ function downloadMultifilesTestTarget(cb) {
+ * If the command fails, the current process exits with code 1.
+ * @param {string} title The title of the command in the hyperfine output.
+ * @param {string} command The command to run.
  * @returns {void}
  */
-function runPerformanceTest(title, targets, multiplier, cb) {
-	const cpuSpeed = os.cpus()[0].speed,
-		max = multiplier / cpuSpeed,
-		cmd = `${ESLINT}--config "${PERF_ESLINTRC}" --no-config-lookup --no-ignore ${targets}`;
-
-	echo("");
-	echo(title);
-	echo("  CPU Speed is %d with multiplier %d", cpuSpeed, multiplier);
-
-	time(cmd, 5, 1, [], results => {
-		if (!results || results.length === 0) {
-			// No results? Something is wrong.
-			throw new Error("Performance test failed.");
-		}
-
-		results.sort((a, b) => a - b);
-
-		const median = results[~~(results.length / 2)];
-
-		echo("");
-		if (median > max) {
-			echo(
-				"  Performance budget exceeded: %dms (limit: %dms)",
-				median,
-				max,
-			);
-		} else {
-			echo("  Performance budget ok:  %dms (limit: %dms)", median, max);
-		}
-		echo("");
-		cb();
-	});
+function runPerformanceTest(title, command) {
+	// We don't need the stack trace of execFileSync if the command fails.
+	try {
+		/*
+		 * The used hyperfine options are:
+		 *   --shell=none turns off the shell escaping, so that glob patterns are not expanded.
+		 *   --warmup=1 runs the command once before measuring, to avoid cold start issues.
+		 *   --runs=5 runs the command 5 times, not counting the warmup run.
+		 *   --command-name sets the title of the command in the hyperfine output.
+		 *
+		 * The ANSI escape codes are used to overwrite the text "Benchmark 1: " that hyperfine prints by default,
+		 * and to set the title in bold.
+		 *   `\x1b[1K` clears the line
+		 *   `\x1b[99D` moves the cursor back to the beginning of the line
+		 *   `\x1b[1m` sets the text to bold
+		 *   `\x1b[0m` resets the text formatting
+		 */
+		childProcess.execFileSync(
+			"hyperfine",
+			[
+				"--shell=none",
+				"--warmup=1",
+				"--runs=5",
+				"--command-name",
+				`\x1b[1K\x1b[99D\x1b[1m${title}\x1b[0m`,
+				command,
+			],
+			{ stdio: "inherit" },
+		);
+	} catch {
+		exit(1);
+	}
 }
 
-/**
- * Run the load performance for eslint
- * @returns {void}
- * @private
- */
-function loadPerformance() {
-	echo("");
-	echo("Loading:");
+target.perf = () => {
+	checkHyperfineInstallation();
 
-	const results = [];
+	downloadMultifilesTestTarget();
 
-	for (let cnt = 0; cnt < 5; cnt++) {
-		const loadPerfData = loadPerf({
-			checkDependencies: false,
-		});
+	createConfigForPerformanceTest();
 
-		echo(
-			`  Load performance Run #${cnt + 1}:  %dms`,
-			loadPerfData.loadTime,
-		);
-		results.push(loadPerfData.loadTime);
-	}
+	// Empty line for better readability in the console output.
+	console.log();
 
-	results.sort((a, b) => a - b);
-	const median = results[~~(results.length / 2)];
+	const loadingCommand = `"${process.execPath}" --require "${require("./package.json").main}" ""`;
```

Comment:
Question for better understanding: what is the expected purpose of the `""` at the end of the command? When I run the command (`"C:\Program Files\nodejs\node.exe" --require "./lib/api.js" ""`) directly, Node.js prompts for input. Perhaps it was intended to be `--eval ""`? Though, I don't see any difference between `""` present, `""` omitted, and `--eval ""` when running `npm run test:performance`.